dashboard/test/models/user_test.rb

```diff
@@ -4140,43 +4140,46 @@ def hide_scripts_in_sections(section1, section2)
     assert_equal teacher.user_school_infos.count, 2
   end
 
-  test 'does set admin to true when google oauth codeorg account' do
+  test 'can grant admin role with google oauth, codeorg account' do
     email = 'fernhunt@code.org'
     migrated_teacher = create(:teacher, :with_google_authentication_option, email: email)
 
-    migrated_teacher.admin = true
-    migrated_teacher.save!
+    migrated_teacher.update!(admin: true)
 
-    assert migrated_teacher.admin
+    assert migrated_teacher.valid?
+    assert migrated_teacher.errors[:admin].empty?
   end
 
-  test 'does set admin to false when it is not a google oauth codeorg account' do
+  test 'cannot grant admin role when google authentication option not does exist' do
     email = 'annieeasley@code.org'
     migrated_teacher = create(:teacher, email: email)
 
-    migrated_teacher.admin = true
-    migrated_teacher.save!
+    assert_raises(ActiveRecord::RecordInvalid) do
+      migrated_teacher.update!(admin: true)
+    end
 
-    refute migrated_teacher.admin
+    refute migrated_teacher.reload.admin?
+    assert migrated_teacher.errors[:admin].length == 2
```

Comment:
i think it would be useful to add an assertion that is more explicit about the errors it expects. something like:

```
assert_equal ["you can't do that!"],  migrated_teacher.errors.full_messages
```

just to be extra safe that not only do we have the correct number of errors, we also have the exact errors we expect

Reply:
(same goes for the similar tests below this one 😃 )

Reply:
@madelynkasula tests updated.  thanks!